content/docs/guides/gcp-e2e.md

```diff
@@ -0,0 +1,737 @@
+the range 0 to 9, as well as the labels identifying the digit in each image.
+
+After training, the model classifies incoming images into 10 categories (0 to 9)
+based on what it's learned about handwritten images. In other words, you send
+an image to the model, and the model does its best to identify the digit shown
+in the image.
+
+TODO(sarahmaddox): Add screenshot of the prediction UI - same as shown at end of tutorial.
+
+TODO(sarahmaddox): Add workflow diagrams at the relevant stages in this tutorial.
+
+### The overall workflow
+
+Here's an overview of what you accomplish by following this guide:
+
+* Set up [Kubeflow][kubeflow] in a [Kubernetes Engine][kubernetes-engine]
+  cluster.
+
+* Test the code locally using a [Jupyter notebook][jupyterhub].
+
+* Train the model:
+
+  * Package a TensorFlow program in a [Kubernetes][kubernetes] container.
+  * Upload the container to [Container Registry][container-registry].
+  * Submit a [tf.train][tf-train] job.
+
+* Use the model for prediction (inference):
+
+  * Save the trained model to [Cloud Storage][cloud-storage].
+  * Use [TensorFlow Serving][tf-serving] to serve the model.
+  * Build a simple web app to send a prediction request to the model and display
+    the result.
+
+## Setup
+
+### Download the project files
+
+TODO(sarahmaddox): Update this section when the updated sample is in the Kubeflow repo.
+
+TODO(sarahmaddox): Also change all mentions of directory "kubeflow-introduction" to "kubeflow-gcp" or something similar, depending on repo name.
+
+TODO(sarahmaddox): Add distributed training to the MNIST model (Python code and YAML files).
+
+To simplify this tutorial, you can use a set of prepared files that include
+a TensorFlow application for training your model, a web UI to send prediction
+requests and display the results, and the [Docker][docker] files to build
+runnable containers for the training and prediction applications.
+
+Download the project files and then go to the directory thus created:
+
+```
+cd ${HOME}
+git clone https://github.com/googlecodelabs/kubeflow-introduction
+cd kubeflow-introduction
+```
+
+### Set up your GCP account and SDK
+
+Follow these steps to set up your GCP environment:
+
+1. Select or create a project on the [GCP Console][gcp-console].
+1. Make sure that billing is enabled for your project. See the guide to
+  [modifying a project's billing settings][billing-guide].
+1. Install the [Cloud SDK][cloud-sdk].
+
+Notes:
+
+* As you work through this tutorial, your project uses billable components of
+  GCP. To minimise costs, follow the instructions to
+  [clean up resources](#cleanup) when you've finished with them.
+* This guide assumes you want to manage your GCP environment on your own server
+  rather than in the [Cloud Shell][cloud-shell] environment. If you choose to
+  use the Cloud Shell, you'll notice that some of the components are
+  pre-installed in your shell. That's OK.
+
+### Create OAuth client credentials
```

Comment:
Good point. I'd like to finish this doc as an end-to-end guide, then decide whether we should adjust the content in the GKE getting-started guide and/or link to the GKE getting-started guide.

As a general principle, it's better not to force readers to hop off onto another page, because (a) that's a disturbing UX, and (b) the other page may change and become inconsistent with this one. We need to weigh up the maintenance concerns with the UX. 

I also plan to look into some sort of content reuse system for Hugo, to mitigate this sort of problem. Ideas welcome.

Reply:
Not an expert on Hugo, but maybe look into using shortcodes? https://www.giraffeacademy.com/static-site-generators/hugo/shortcode-templates/

Reply:
Thanks, those look about right. Similar to Django, which I use daily.